### PR TITLE
refactor(render): use `RenderElementRef` in all renderer methods

### PR DESCRIPTION
**Pull request by tbosch**
BREAKING CHANGE:
- Almost all methods in `Renderer` now take a `RenderElementRef` instead
  of a `ViewRef` + `boundElementIndex`.
- These methods can be called with the `ElementRef` from the app side
  directly.

### CODE REVIEW

File: modules/angular2/src/core/compiler/element_injector.ts [hunk 1]
```diff
@@ -771,9 +771,7 @@ export class ElementInjector extends TreeNode<ElementInjector> {
 
   getComponent(): any { return this._strategy.getComponent(); }
 
-  getElementRef(): ElementRef {
-    return new ElementRef(new ViewRef(this._preBuiltObjects.view), this._proto.index);
-  }
+  getElementRef(): ElementRef { return this._preBuiltObjects.view.elementRefs[this._proto.index]; }
```
Comment (rkirov) [inline modules/angular2/src/core/compiler/element_injector.ts]: In general why not add the ref to prebuiltObjects, I don't see value in going through view.

File: modules/angular2/src/core/compiler/view.ts [hunk 2]
```diff
@@ -40,6 +42,8 @@ export class AppView implements ChangeDispatcher, EventDispatcher {
   componentChildViews: List<AppView> = null;
   viewContainers: List<AppViewContainer>;
   preBuiltObjects: List<PreBuiltObjects> = null;
+  elementRefs: List<ElementRef>;
+  ref: ViewRef;
```
Comment (rkirov) [inline modules/angular2/src/core/compiler/view.ts]: Do we still need render?

File: modules/angular2/src/core/compiler/view_container_ref.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ export class ViewContainerRef {
     }
   }
 
-  get(index: number): ViewRef { return new ViewRef(this._getViews()[index]); }
+  get(index: number): ViewRef { return this._getViews()[index].ref; }
```
Comment (rkirov) [inline modules/angular2/src/core/compiler/view_container_ref.ts]: +1 there should be no way to create two refs to the same view.

File: modules/angular2/test/core/compiler/element_injector_spec.ts [hunk 4]
```diff
@@ -59,14 +58,25 @@ import {QueryList} from 'angular2/src/core/compiler/query_list';
...
   changeDetector;
-  constructor() {
-    super();
+  elementRefs;
+  constructor(elementCount = 0) {
+    super(AppView);
     this.componentChildViews = [];
     this.changeDetector = null;
+    this.elementRefs = ListWrapper.createFixedSize(elementCount);
+    for (var i=0; i<elementCount; i++) {
```
Comment (rkirov) [inline modules/angular2/test/core/compiler/element_injector_spec.ts]: white-space `var i = 0; i < elementCount`

File: modules/angular2/src/core/compiler/element_injector.ts (hunk 1 above)
Comment (tbosch) [inline modules/angular2/src/core/compiler/element_injector.ts]: We need the view already for other purposes.

**Pull request closed without merging**